com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java

```diff
@@ -727,7 +728,13 @@ public void setFormattingActive(final boolean formattingActive) {
   private boolean isCommentNode(final INode node) {
     if (node instanceof ILeafNode && node.getGrammarElement() != null) {
       EObject grammarElement = node.getGrammarElement();
-      return grammarElement.equals(formatter.getSLCommentRule()) || grammarElement.equals(formatter.getMLCommentRule());
+      if (grammarElement instanceof TerminalRule) {

```

Comment:
must always be a TerminalRule? Should we not keep the old code in an else branch?

Reply:
The old code always returns false for non-TerminalRule elements as the comment rules are TerminalRules.